Rename CalculateMultipleScattering algorithm

DannyHindson opened the pull request:
**Description of work.**

Rename CalculateMultipleScattering to DiscusMultipleScatteringCorrection to avoid confusion with the multiple scattering algorithm that has been written by SNS that uses numerical integration (MultipleScatteringCorrection). The word Discus refers to the original Fortran program written by Mike Johnson where the code came from.

There were a couple of other approaches that I considered but decided against in the end:
- adding the words MonteCarlo to the name. There is already another Monte Carlo multiple scattering correction in Mantid (Mayers) so this didn't seem suitable. Adding further words (eg NonIsotropic) to the name to further describe the functionality didn't seem practical
- the approach of having the Calculate prefix didn't seem to be as prevalent in the sample correction algorithms as simply ending the algorithm name with XXXCorrection

**To test:**

Code review only

Fixes #32289.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

DannyHindson committed:
Rename CalculateMultipleScattering

Rename CalculateMultipleScattering to CalculateMonteCarloMultipleScattering
to avoid confusion with the multiple scattering algorithm that has been
written by SNS that uses numerical integration (MultipleScatteringCorrection)

DannyHindson committed:
Update release notes for algorithm rename

DannyHindson committed:
Change name of ms algorithm again and fix up concepts page plots

DannyHindson commented:
@peterfpeterson - I've put this PR in to rename the multiple scattering algorithm I created. I've renamed it to "DiscusMultipleScatteringCorrection" in the end after the original Fortran program because there's already another Monte Carlo multiple scattering correction (MayersSampleCorrection)

I haven't waited for a broader discussion on how to name the correction algorithms because we're pretty much at the point where we need to get something into the release. If there's a future discussion then I can change again. The algorithm already generates a warning saying the name\parameters may change.

As part of this I've modified the comparison plots on the Concepts page to take into account the new name. I noticed there were a couple of glitches in the python script (abs and ms back to front in some places + radius parameter missing on Carpenter correction) so I've fixed these at the same time.

peterfpeterson commented on docs/source/concepts/AbsorptionAndMultipleScattering.rst:
```diff
@@ -697,7 +697,10 @@ Multiple Scattering Correction Algorithms in Mantid Table
 |                                                                                     |             |            |                                 |                      |                     ||  In-plane only                                                                       |
 +-------------------------------------------------------------------------------------+-------------+------------+---------------------------------+----------------------+---------------------+---------------------------------------------------------------------------------------+
-| :ref:`CalculateMultipleScattering <algm-CalculateMultipleScattering>`               | E           | MC         | Any Shape                       | Wavelength           | EA, FI, PI          ||  Based on Monte Carlo program DISCUS [16]_ (or MINUS). [17]_                         |
+| :ref:`CarpenterSampleCorrection <algm-CarpenterSampleCorrection>`                   | E           | NI         | Cylinder                        | Wavelength           | IA,EA,FI,W          ||  Only applicable to Vanadium                                                         |
```
Carpenter is actually an approximation using spherical Bessel functions. It should be labeled as analytical. This is similar to Mayers which uses a Chebyshev polynomial to approximate things.

DannyHindson replied:
OK. I didn't change this in the PR but I can change the NI to A while I'm doing a few changes in this file

DannyHindson committed:
Relabel Carpenter multiple scattering algorithms as analytical

DannyHindson committed:
Clarify description of Monte Carlo techniques

Following discussion with ORNL team, modify description of Monte Carlo
integration and Monte Carlo ray tracing to indicate there is some
overlap between these techniques

DannyHindson commented:
@peterfpeterson - I've had a go at rewording the description of the MC integration and ray tracing definitions. Rather than just removing them I've instead suggested they overlap. In the tables that contain the different classifications of the algorithms I've simplified the list to Analytical, Numerical Integration and simply "Monte Carlo"

sf1919 commented:
To be merged after #32559 .

gemmaguest commented:
@DannyHindson Can I just check that some functional testing has been done, as this touches functional code? Even if it's just to check that the algorithm still runs...

DannyHindson commented:
> @DannyHindson Can I just check that some functional testing has been done, as this touches functional code? Even if it's just to check that the algorithm still runs...

The algorithm has a unit test which runs it using the new name but just to be doubly sure I've just run the following script to be sure and it runs fine:
```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

def loadPEARLWSAndRunMS(SpectrumList, OutputName):
    inputWS = Load(Filename=os.path.join(folder,'PEARL00115245.nxs'), SpectrumList=SpectrumList) # Ce02
    # S(Q) has been cropped 1500 to 19995 during PEARL data reduction so do same here
    inputWS = CropWorkspace(inputWS, 1500, 19995)
    inputWS = ConvertUnits(inputWS, Target="Wavelength")

    # 5mm x 5mm beam
    SetBeam(inputWS, Geometry={'Shape': 'Slit', 'Width': 0.5, 'Height': 0.5})
    # 6mm diameter can
    SetSample(InputWorkspace=inputWS,Geometry={'Shape': 'Cylinder', 'Height': 4.0,
                        'Radius': 0.2985, 'Center': [0.,0.,0.]},
              Material={'ChemicalFormula': 'Ce-O2', 'MassDensity': 7.215})
           
    DiscusMultipleScatteringCorrection(InputWorkspace=inputWS,SofqWorkspace=SOfQ,
           NeutronPathsSingle=100,NeutronPathsMultiple=100, NumberScatterings=3, 
           OutputWorkspace=OutputName, Interpolation="Linear", NumberOfWavelengthPoints=300)

folder = r"C:\Multiple Scattering\PEARL tests"
SOfQ = Load(Filename=os.path.join(folder,'PRL115245-115251_tt70_mods1-9_Q.nxs')) #Ce02
# scale S(Q) so it tends to 1 at large q - rough approach for now
high_q = SOfQ.dataY(0)[-9:]
high_q_avg = sum(high_q)/len(high_q)
SOfQ_y = SOfQ.dataY(0)
SOfQ_y /= high_q_avg

SOfQ_x = SOfQ.dataX(0)
average_q_spacing = (SOfQ_x[-1]-SOfQ_x[0])/(SOfQ_x.size-1)
SOfQ=Rebin(SOfQ,[SOfQ_x[0],average_q_spacing,SOfQ_x[-1]])

loadPEARLWSAndRunMS(20, "MuscatResultsSp20")
```

DannyHindson commented:
@peterfpeterson - I fixed a merge conflict earlier which dismissed you review. Please could you re-approve?
